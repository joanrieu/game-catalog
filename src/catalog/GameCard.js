import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Card = styled.section`
  min-width: 320px;
  max-width: 350px;
  background: #c4c4c4;
  border-radius: 8px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  overflow: hidden;
`;

const Screenshot = styled.img`
  height: 411px;
`;

const Description = styled.div`
  padding: 16px;
  background: #ffffff;
`;

const Name = styled.h2`
  margin: 0;
  padding: 0;
  font-weight: normal;
  font-size: 18px;
`;

const Platform = styled.p`
  margin: 0;
  margin-top: 8px;
  padding: 0;
  font-size: 14px;
`;

export default function GameCard({ title, platform, screenshot }) {
  return (
    <Card>
      <Screenshot src={screenshot} />
      <Description>
        <Name>{title}</Name>
        <Platform>{platform}</Platform>
      </Description>
    </Card>
  );
}

GameCard.propTypes = {
  title: PropTypes.string.isRequired,
  platform: PropTypes.string.isRequired,
  screenshot: PropTypes.string
};
