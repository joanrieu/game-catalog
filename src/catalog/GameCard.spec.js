import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import GameCard from "./GameCard";

describe("GameCard", () => {
  it("has a screenshot, a title and a platform", () => {
    const cmp = renderer.create(
      <GameCard
        title="Title"
        platform="PS4"
        screenshot="//placehold.it/300x300"
      />
    );
    expect(cmp).toMatchSnapshot();
  });
});
