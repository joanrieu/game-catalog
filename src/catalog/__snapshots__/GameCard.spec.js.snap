// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard has a screenshot, a title and a platform 1`] = `
.c0 {
  min-width: 320px;
  max-width: 350px;
  background: #c4c4c4;
  border-radius: 8px;
  box-shadow: 0px 4px 4px rgba(0,0,0,0.25);
  overflow: hidden;
}

.c1 {
  height: 411px;
}

.c2 {
  padding: 16px;
  background: #ffffff;
}

.c3 {
  margin: 0;
  padding: 0;
  font-weight: normal;
  font-size: 18px;
}

.c4 {
  margin: 0;
  margin-top: 8px;
  padding: 0;
  font-size: 14px;
}

<section
  className="c0"
>
  <img
    className="c1"
    src="//placehold.it/300x300"
  />
  <div
    className="c2"
  >
    <h2
      className="c3"
    >
      Title
    </h2>
    <p
      className="c4"
    >
      PS4
    </p>
  </div>
</section>
`;
